planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/clothoid_pull_out.cpp

```diff
@@ -0,0 +1,1680 @@
+
+  double target_dx = target_end.x - target_start.x;
+  double target_dy = target_end.y - target_start.y;
+  double target_length = std::sqrt(target_dx * target_dx + target_dy * target_dy);
+
+  // Calculate scaling factor
+  double scale_factor = (clothoid_length > 1e-10) ? target_length / clothoid_length : 1.0;
+
+  // Calculate rotation angle
+  double clothoid_angle = std::atan2(clothoid_dy, clothoid_dx);
+  double target_angle = std::atan2(target_dy, target_dx);
+  double rotation_angle = target_angle - clothoid_angle;
+
+  // Normalize angle to [-π, π] range
+  while (rotation_angle > M_PI) rotation_angle -= 2 * M_PI;
+  while (rotation_angle < -M_PI) rotation_angle += 2 * M_PI;
+
+  // Choose shorter rotation if over 180 degrees
+  if (std::abs(rotation_angle) > M_PI) {
+    rotation_angle = (rotation_angle > 0) ? rotation_angle - 2 * M_PI : rotation_angle + 2 * M_PI;
+  }
+
+  // Calculate transformation matrix elements
+  double cos_theta = std::cos(rotation_angle);
+  double sin_theta = std::sin(rotation_angle);
+
+  // Apply rigid transformation
+  std::vector<geometry_msgs::msg::Point> corrected_points;
+  corrected_points.reserve(clothoid_points.size());
+
+  for (size_t i = 0; i < clothoid_points.size(); ++i) {
+    geometry_msgs::msg::Point corrected_point;
+
+    // Move to origin
+    double rel_x = clothoid_points[i].x - clothoid_start.x;
+    double rel_y = clothoid_points[i].y - clothoid_start.y;
+
+    // Scale
+    rel_x *= scale_factor;
+    rel_y *= scale_factor;
+
+    // Rotate
+    double rotated_x = cos_theta * rel_x - sin_theta * rel_y;
+    double rotated_y = sin_theta * rel_x + cos_theta * rel_y;
+
+    // Translate to target start
+    corrected_point.x = rotated_x + target_start.x;
+    corrected_point.y = rotated_y + target_start.y;
+    corrected_point.z = clothoid_points[i].z;  // Keep Z coordinate as is
+
+    corrected_points.push_back(corrected_point);
+  }
+
+  return corrected_points;
+}
+
+std::pair<std::vector<geometry_msgs::msg::Pose>, geometry_msgs::msg::Pose>
+generate_clothoid_entry_with_yaw(
+  const ClothoidSegment & segment, const geometry_msgs::msg::Pose & start_pose, int num_points)
+{
+  double A = segment.A;
+  double L = segment.L;
+  double direction_factor = segment.is_clockwise ? -1.0 : 1.0;
+  double start_yaw = tf2::getYaw(start_pose.orientation);
+
+  std::vector<geometry_msgs::msg::Pose> poses;
+
+  // Entry Clothoid: linearly increase curvature from 0 to target curvature
+  double target_curvature = (L / (A * A)) * direction_factor;
+  double start_curvature = 0.0;
+
+  // Accurate calculation using numerical integration
+  double current_x = start_pose.position.x;
+  double current_y = start_pose.position.y;
+  double current_psi = start_yaw;
+
+  for (int i = 0; i < num_points; ++i) {
+    // Create current point and add to poses
+    geometry_msgs::msg::Pose pose;
+    pose.position.x = current_x;
+    pose.position.y = current_y;
+    pose.position.z = 0.0;
```

Comment:
Is `z` set somewhere?

Reply:
same to another part

Reply:
ah, here?
```
    // Set z coordinate: get z value of closest point from lanelet information
```

Reply:
@kosuke55 
> ah, here?

yes
I was thinking that temporarily embedding 0 and ultimately set the z-coordinate properly shouldn't be a problem, but could this cause issues during resampling or orientation calculations?

If it won't be a problem, I'm planning to add comments.